### PR TITLE
Add testing for the vLLM python model

### PR DESCRIPTION
**Pull request by pskiran1**
Changes:

- Added testing for Python backend vLLM model
- Added pre-commit hooks

### CODE REVIEW

**Comment by dyastremsky** [comment]
Great work and tests! Based on the approach that Triton uses elsewhere, I think these tests should be in the server repo's `qa` folder with accompanying changes in GitLab to automatically run these tests. This PR replicating a lot of code in that repository, and that's the entry point for running tests for all backends at present.

Similarly, I think cloning the tutorials repo and testing them directly would be a better approach. If you copy the code over, it won't actually be testing the tutorial, since the code here will be out of step with the

**Comment by dyastremsky** [comment]
> Great work and tests! Based on the approach that Triton uses elsewhere, I think these tests should be in the server repo's `qa` folder with accompanying changes in GitLab to automatically run these tests. This PR replicating a lot of code in that repository, and that's the entry point for running tests for all backends at present.
> 
> Similarly, I think cloning the tutorials repo and testing them directly would be a better approach. If you copy the code over, it won't actually be testing the tutorial, since the code here will be out of step with the tutorial code.

Discussed offline with @tanmayv25. We are working to test the backend code for now, then we can integrate this testing further in a future release.

**Review by dyastremsky — APPROVED** [review]

File: ci/qa_models/vllm/model.py
Comment (tanmayv25) [inline ci/qa_models/vllm/model.py:1]: This should not be needed. The model should be installed as a backend in the container. Am I missing something?

File: ci/qa_models/vllm/config.pbtxt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+name: "vllm"
+backend: "python"
```
Comment (tanmayv25) [inline ci/qa_models/vllm/config.pbtxt:28]: We should directly place vllm in the backend field.

**Comment by pskiran1** [comment]
Closing this PR, will add testing for the tutorials separate from the backend repo.

**Pull request closed without merging**